src/pages/settings/Profile/CustomStatus/StatusClearAfterPage.js

```diff
@@ -0,0 +1,236 @@
+import React, {useEffect, useState, useCallback, useMemo} from 'react';
+import {View} from 'react-native';
+import _ from 'lodash';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import ScreenWrapper from '../../../../components/ScreenWrapper';
+import Form from '../../../../components/Form';
+import HeaderWithBackButton from '../../../../components/HeaderWithBackButton';
+import ROUTES from '../../../../ROUTES';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import styles from '../../../../styles/styles';
+import Text from '../../../../components/Text';
+import MenuItemWithTopDescription from '../../../../components/MenuItemWithTopDescription';
+import SelectionList from '../../../../components/SelectionList';
+import useLocalize from '../../../../hooks/useLocalize';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import CONST from '../../../../CONST';
+import * as User from '../../../../libs/actions/User';
+import withLocalize from '../../../../components/withLocalize';
+import compose from '../../../../libs/compose';
+import DateUtils from '../../../../libs/DateUtils';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps} from '../../../../components/withCurrentUserPersonalDetails';
+import personalDetailsPropType from '../../../personalDetailsPropType';
+
+const defaultProps = {
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+const propTypes = {
+    currentUserPersonalDetails: personalDetailsPropType,
+    customStatus: PropTypes.shape({
+        clearAfter: PropTypes.string,
+        customDateTemporary: PropTypes.string,
+    }),
+};
+
+/**
+ * @param {string} data - one of the values from CONST.CUSTOM_STATUS_TYPES or data in format YYYY-MM-DD HH:mm
+ * @returns {string}
+ */
+function getInitialDateBasedFromStatusType(data) {
```

Comment:
Could we make this a bit clearer? `getSelectedStatusType` or something similar.

Reply:
👍🏻